nms/app/packages/magmalte/app/views/equipment/GatewayDetailConfigEdit.js

```diff
@@ -291,9 +262,7 @@ function GatewayEditDialog(props: DialogProps) {
       {tabPos === 1 && (
         <DynamicServicesEdit
           isAdd={!editProps}
-          gateway={
-            Object.keys(gateway).length != 0 ? gateway : ctx.state[gatewayId]
-          }
+          gateway={ctx.state[gatewayId] || gateway}
```

Comment:
shouldn't it be gateway || ctx.state[gatewayId] rather ? Otherwise we will be overriding recent values with context values ?